paprika/analyze.py

```diff
@@ -18,69 +17,112 @@ class Analyze(object):
-                        not restraint.mask4:
-                    r_restraint = restraint
-                if restraint.mask3 and not restraint.mask4:
-                    mask3_residue_name = structure[restraint.mask3].residues[0].name
-                    if "DM" in restraint.mask1 and "DM" in restraint.mask2 and self.guest_residue_name.upper() in mask3_residue_name:
-                        theta_restraint = restraint
-                    if "DM" in restraint.mask1 and self.guest_residue_name.upper() in mask2_residue_name and self.guest_residue_name.upper() in \
-                            mask3_residue_name:
-                        beta_restraint = restraint
-            if r_restraint and theta_restraint and beta_restraint:
-                analysis.compute_ref_state_work(
-                    [r_restraint, theta_restraint, None, None, beta_restraint, None]
-                )
-            else:
-                logging.warning("Could not determine the r, θ, or β restraint for computing the analytic release step.")
-
-        else:
-            analysis.compute_free_energy(phases=["release"])
-
-        return analysis
+        analysis.temperature = temperature
+
+        distance_restraint = next(
+            (
+                restraint
+                for restraint in guest_restraints
+                if "DM" in restraint.mask1
+                and restraint.mask2 is not None
+                and restraint.mask3 is None
+                and restraint.mask4 is None
+            ),
+            None
+        )
+
+        theta_restraint = next(
+            (
+                restraint
+                for restraint in guest_restraints
+                if "DM" in restraint.mask1
+                and "DM" in restraint.mask2
+                and restraint.mask3 is not None
+                and restraint.mask4 is None
+            ),
+            None
+        )
+        beta_restraint = next(
+            (
+                restraint
+                for restraint in guest_restraints
+                if "DM" in restraint.mask1
+                and "DM" not in restraint.mask2
+                and restraint.mask3 is not None
+                and restraint.mask4 is None
+            ),
+            None
+        )
+
+        if not distance_restraint or not theta_restraint or not beta_restraint:
+
+            raise RuntimeError(
+                "Could not determine the r, θ, or β restraint for computing the "
+                "analytic release step."
+            )
+
+        analysis.compute_ref_state_work(
+            [distance_restraint, theta_restraint, None, None, beta_restraint, None]
+        )
+
+        return analysis.results["ref_state_work"]
+
+    @classmethod
+    def symmetry_correction(cls, microstates: int) -> float:
+
+        assert microstates > 0
+
+        if microstates == 1:
+            return 0.0
+
+        return -0.593 * np.log(microstates)
```

Comment:
maybe have temperature as an input argument to this method so it returns the energy with the correct temperature (if a temperature other than 298.15 K is used)

Reply:
Yeah, good point.

Also, @SimonBoothroyd do you think we should import `units` here and set this to be in kcal/mol rather than leaving it to be implicit?

Reply:
Added.

> Also, @SimonBoothroyd do you think we should import units here and set this to be in kcal/mol rather than leaving it to be implicit?

Hmm units are a bit tricky because there are several packages which provide them that we could go with (e.g. `simtk` but this ties things more to OpenMM, `pint` but then units are attached to a particular unit registry and aren't easily transferred between packages etc...). May be best to leave it for now and perhaps introduce in a separate PR?

Reply:
Exactly my thinking.